### PR TITLE
Some map loader and vault stuff + asteroid pond vault update

### PR DESCRIPTION
**Pull request by unid15**
- It's now possible to load a vault/other map element without specifying any coordinates, the game will just put it in a good secure place on z2. You can load a lot of map elements this way, and they will never overlap.
- Added a proc "load_dungeon( _map element object or path_ )" that does that
- "Load Map Element" admin verb now contains this option too
- You need to put a marker (/obj/effect/landmark/dungeon_area) somewhere on z2 for this feature to work

This can be done to make more interesting vaults. For example, a wizard's den that contains a portal to another dimension. The "another dimension" would be a 60x60 map hidden on z-level 2, loaded only after the main vault spawns.

As an example of a two-map vault, I added a wine cellar to the asteroid pond. Code-wise, it's extremely simple
pond
The ladder leads to

### CODE REVIEW

**Comment by ihadtoregisterforthis** [comment]
Holy moley, take a rest

**Comment by ghost** [comment]
So, the dungeon marker would need to be added to each map with its own containment zone?

What happens to the frog pond if there's no dungeon markers on the map it would load the dungeon on?

**Comment by unid15** [comment]
Yep, all maps will have to be modified to utilize this feature

If there's no dungeon marker, the dungeon part will simply not load. So there will just be a frog pond with a non-functioning ladder.
I also added a variable for vaults that prevents them from spawning in maps without a dungeon marker, for those vaults where the dungeons are more important

**Comment by ghost** [comment]
Could you potentially have a dungeon ladder within a dungeon?

**Comment by unid15** [comment]
Sure, with some coding

**Comment by unid15** [comment]
Ready

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -1142,8 +1142,17 @@ var/list/admin_verbs_mod = list(
 	var/y_coord
 	var/z_coord
 
-	switch(alert(usr, "Select a location for the new map element", "Map element loading", "Use my current location", "Input coordinates", "Cancel"))
-		if("Use my current location")
+	#define ML_CURRENT_LOC  "Use my current location"
+	#define ML_INPUT_COORDS "Input coordinates"
+	#define ML_LOAD_TO_Z2   "Find a suitable location at Z-level 2 (done automatically)"
+	var/list/choices = list(
```
Comment (PJB3005) [inline code/modules/admin/admin_verbs.dm]: Make this a static so it doesn't rebuild the list every time.

**Pull request merged**